### ISSUE TITLE
How are “new” column names in input files generated?

### ISSUE THREAD

**Issue by leewujung** [OSOceanAcoustics/echopop#155]
The input files have gone through a round of changes to bring all the input column names into all lower case standardized names. However judging from a recent case that a column name had to be manually updated to conform with what’s expected in Python EchoPro, I’m not sure how these column name changes were implemented.

@emiliom : Could you summarize the current status re the following:
- Are these columns going to be generated in the new naming format from FEAT database?
- If answer to the above is No, what was the agreement? 

Thanks!

**Comment by emiliom** [comment]
**TEMPORARY: I'M SAVING THIS COMMENT ONCE BEFORE COMPLETING, TO BE ON THE SAFE SIDE! I'M JUST SAVING IT AND CONTINUING TO WORK ON IT**

I'll discuss these in two broad categories:

### 1. Biological data files
Biological files contain data from trawls and are the ones that come from the FEAT relational database that Alicia maintains (and Emilio received a database dump for local R&D). See the [input-files docs page](https://uw-echospace.github.io/EchoPro/input-files.html#biological-trawl-data). Alicia and I invested a good chunk of effort to create the SQL queries that can lead to the Excel input files with the minimum information needed by EchoPro. The query for the length file is discussed [here](https://github.com/uw-echospace/hake-IGP/issues/2), the one for the specimen file [here](https://github.com/uw-echospace/hake-IGP/issues/3), and the one for the haul-vs-transect mapping [here](https://github.com/uw-echospace/hake-IGP/issues/4). The SQL code was provided by Alicia, then tweaked by Emilio after some back and forth.

We're missing the SQL query code for the catch file. I think that was an oversight in the initial discussions between Brandon and Alicia, which may not have surfaced the catch file early on.

Brandon and Alicia held several meetings and exchanges about the biological data files. I think they settled on the column names presented in the [`echopro_bio_input_template` Google Sheet](https://docs.google.com/spreadsheets/d/180h_kSTnGiBUUMOKK5z_-8Saw0R_8P3HTjUJyyBRxsE/edit#gid=0), which in turn is what's shown in the input-files docs page. I don't think they discussed the catch file, but that file has only one column, `haul_weight`, whose name is not found elsewhere; that name was probably chosen by Brandon and Emilio.

Ultimately, the operational PostgreSQL Views and the [Python code](https://github.com/uw-echospace/hake-IGP/issues/5) that generates the expected Excel input files will be deployed on Alicia's end. 

### 2. All other files

All other input files came from Chu. See the `Stratification`, `NASC` and `Kriging` files in the [input-files docs page](https://uw-echospace.github.io/EchoPro/input-files.html#). 

- Stratification: I don't know how these files are generated.
- NASC: 
- Kriging: These files were likely generated long ago and have likely not changed, or not changed much, from survey to survey. But I don't know for sure.

**Comment by leewujung** [comment]
> * NASC: The NASC file is a "pre-processing" component of Matlab EchoPro that has not been implemented and we were going to implement. It combines the individual EchoView EVL files into the single NASC file used by EchoPro. The code seems to be contained largely, possibly fully, in [general/processing/generate_EV_EI_NASC_table.m](https://github.com/uw-echospace/EchoPro_matlab/blob/main/general/processing/generate_EV_EI_NASC_table.m). Note: A slim issue was created for this task a while ago, [Construction of NASC data files #83](https://github.com/uw-echospace/EchoPro/issues/83).

I will close #83 because there is no information there, and it was incorrect to treat the NASC file as something that FEAT will give us. The inputs to EchoPro are Echoview exports. The script you pointed to is not the actual core of constructing the NASC file. It is this one: https://github.com/uw-echospace/EchoPro_matlab/blob/eb6ac07d48f13c8ebd173b0072f67015bcacf205/general/processing/construct_NASC_int_file_new.m.
I will create a new issue for this.

**Comment by leewujung** [comment]
Below are the belated notes from 2024/03/18 meeting with Alicia:
- Alicia did not think she will change variable names in the database, but will create a dictionary for matching/converting the variable names in the database to what Echopop ingests
  - she will rename the columns using the dictionary when pushing the database outputs to excel files via pandas
- This is because the variable names used in Echopop are not what is in the database, e.g., fork length, total length, etc. (which are much more specific)

Action items:
- Once this code refactoring is done (v0.4.3), ask Alicia for fresh (newly queried) 2017 files
- Alicia will go through the database documentation soon and share with APL folks
